packages/contracts-bedrock/test/fixtures/test-deploy-superchain-out.toml

```
@@ -1,3 +1,16 @@
+[dsi]
```

IMO this naming convention makes sense in the code, but is a bit unclear for the toml files, and I expect that many more people will read the config files than the deploy scripts. I'd suggest something like
 
```suggestion
[superchain]
```

---

Agreed that we shouldn't expose the shorthand convention here. I would suggest keeping the input structure the same as it was, and we can use comments in the output file to differentiate the inputs and outputs. The rationale here is that, as a consumer of an output file, it doesn't really matter whether `x` was an input or output of that step. So for example the output file here could be something like the below:

```toml
# -------- DeploySuperchain Inputs --------
paused = false
requiredProtocolVersion = 1
recommendedProtocolVersion = 2

[roles]
proxyAdminOwner = "0x51f0348a9fA2aAbaB45E82825Fbd13d406e04497"
protocolVersionsOwner = "0xeEB4cc05dC0dE43c465f97cfc703D165418CA93A"
guardian = "0xE5DbA98c65F4B9EB0aeEBb3674fE64f88509a1eC"

# -------- DeploySuperchain Outputs --------
protocolVersionsImpl = "0x5991A2dF15A8F6A256D3Ec51E99254Cd3fb576A9"
protocolVersionsProxy = "0x1d1499e622D69689cdf9004d05Ec547d650Ff211"
superchainConfigImpl = "0xF62849F9A0B5Bf2913b396098F7c7019b51A820a"
superchainConfigProxy = "0xc7183455a4C133Ae270771860664b6B7ec320bB1"
superchainProxyAdmin = "0x2e234DAe75C793f67A35089C9d99245E1C58470b"
```

---

Actually, I think my suggestion will put all outputs under the roles header, so maybe we put them under a `contracts` or `outputs` header. The outputs from all steps are always contract addresses

---

> Actually, I think my suggestion will put all outputs under the roles key
It will yea. 

Good to go with `[outputs]` though.

---

If we didn't have the comments, I'd probably be pushing for some identifier/label to indicate which step of the process the output file is from. But the comments make it clear.

---

> Personally, if I've looking at an output file. 

In general I agree, however I think these output files are more likely to be consumed by other tools (op-deployer, superchain registry) than used directly by humans as a reference. We should still aim to make them as clear and readable as we can though, but for example when using the DeploySuperchain outfile to configure a genesis file, it doesn't matter that `proxyAdminOwner` was an input to `DeploySuperchain.s.sol`, it's still an input for genesis generation regardless

---

Added in: https://github.com/ethereum-optimism/optimism/pull/11975/commits/d11660e77476e00456829dc3a37fb499d65328e9#diff-62b3d04f126e442fa8b92be128f8e5a03d3470e09d65f5f5657c5d0f842eff90R204 

I'm not using `awk` as discussed but instead using `printf` and `cat` both of which behave the same across platform. I personally find it easier to read.